src/shared/payment/services/transaction-helper.ts

```diff
@@ -135,21 +148,31 @@ export class TransactionHelper implements OnModuleInit {
     minFee: number,
     from: Asset | Fiat,
     to: Asset | Fiat,
+    tradingLimit: number | undefined,
   ): Promise<TargetEstimation> {
     const price = await this.priceProviderService.getPrice(from, to);
 
+    const fiatChf = from.name !== 'CHF' ? undefined : await this.fiatService.getFiatByName('CHF');
+    const tradingLimitPrice = from.name !== 'CHF' ? undefined : await this.priceProviderService.getPrice(fiatChf, from);
+
     const percentFeeAmount =
       outputAmount != null ? price.invert().convert((outputAmount * fee) / (1 - fee)) : inputAmount * fee;
     const feeAmount = Math.max(percentFeeAmount, minFee);
 
     const targetAmount = outputAmount != null ? outputAmount : price.convert(Math.max(inputAmount - feeAmount, 0));
     const sourceAmount = outputAmount != null ? price.invert().convert(outputAmount) + feeAmount : inputAmount;
 
+    const currentTradingLimit = tradingLimit ? tradingLimit : Config.defaultDailyTradingLimit;
+    const tradingLimitInSource = !tradingLimitPrice
+      ? currentTradingLimit
+      : tradingLimitPrice.convert(currentTradingLimit * 0.99, -1); // -1% for the conversion
+
     return {
       exchangeRate: this.round(price.price, from instanceof Fiat),
       feeAmount: this.round(feeAmount, from instanceof Fiat),
       estimatedAmount: this.round(targetAmount, to instanceof Fiat),
       sourceAmount: this.round(sourceAmount, from instanceof Fiat),
+      tradingLimit: tradingLimitInSource,
     };
```

Comment:
It's a bit weird to return the `tradingLimit` in source asset in a method called `getTargetEstimation` 🤔